tests/unit-tests/test_nn_model.py

```diff
@@ -62,12 +63,46 @@ def create_dummy_test_graph(dim):
               "n3": th.empty((0,dim)),
               "n4": th.empty((0,dim)),}
 
-    return block, inputs, list(edges.keys())
+    return block, inputs, list(num_src_nodes.keys()), list(edges.keys())
 
+
+def create_dummy_no_indegree_test_graph(dim=16):
+    """ Generate dummy test graph in which some destination nodes have no in-degree.
+    This could happen when either these nodes are target nodes, or these nodes have
+    outdegree to target nodes but no indegree to themselves.
+    """
+    num_src_nodes = {
+        "n0": 1024,             # The first n0 256 nodes are des nodes w/t indegree
+        "n1": 128,              # Half of n1 nodes link to n2 nodes, target nodes
+        "n2": 64                # Target nodes are all in src and dst lists.
+    }
+    num_nodes_dict = {
+        "n0": 256,              # The dst nodes w/t indegree
+        "n1": 64,               # Half of n1 nodes are dst nodes from n0 nodes
+        "n2": 64                # some link from n0, and some from n1
+    }
+
+    edges = {
+    ("n0", "r0", "n1"): (th.arange(256, 1024), th.concat([th.arange(64), th.randint(0, 64, (704,))])),
+    ("n0", "r1", "n2"): (th.arange(256), th.concat([th.arange(64), th.randint(0, 64, (192,))])),
+    ("n1", "r2", "n2"): (th.arange(64, 128), th.arange(64))
+    }
+
+    block = dgl.create_block(edges,
+        num_src_nodes=num_src_nodes,
+        num_dst_nodes=num_nodes_dict)
+
+    inputs = {"n0": th.zeros((1024, dim)),
```

Comment:
So in this case n0 does not have in edges, right?

Reply:
only the first 256 n0 nodes do not have in edges. The other 768 n0 nodes have.